compiler-lib/src/lowering/lir.rs

```diff
@@ -0,0 +1,1026 @@
+    }
+}
+
+impl Ptr<ControlFlowTransfer> {
+    /// Only call this from target
+    fn add_incoming_value_flow(mut self, target_slot: Ptr<ValueSlot>, alloc: &Allocator) {
+        if let Some((source, target)) = self
+            .register_transitions
+            .iter()
+            .find(|(_, existing_slot)| target_slot.firm == existing_slot.firm)
+        {
+            log::debug!(
+                "\t\t? {:?}({:?}) := {:?}",
+                target_slot.allocated_in.firm,
+                target.num,
+                target.firm,
+            );
+            for multislot in target.allocated_in.regs.iter() {
+                log::debug!(
+                    "\t\t! {:?}({:?}) := {:?}",
+                    target.allocated_in.firm,
+                    multislot.num(),
+                    multislot.firm(),
+                );
+                if let MultiSlot::Multi { slots, .. } = &**multislot {
+                    for slot in slots.iter() {
+                        log::debug!("\t\t\t {:?} := {:?} @ {:?}", slot.num, slot.firm, slot);
+                    }
+                }
+            }
+            log::debug!(
+                "\t\t> {:?}({:?}) := {:?} @ {:?}",
+                target_slot.allocated_in.firm,
+                target_slot.num,
+                target_slot.firm,
+                target_slot,
+            );
+
+            log::debug!(
+                "\tPIGGY: from='{:?}' to='{:?}' value='{:?}'",
+                source.allocated_in().firm,
+                target.allocated_in.firm,
+                target.firm,
+            );
+            assert_eq!(target.num, target_slot.num);
+
+            return;
+        }
+
+        let source_slot = self.source.new_forwarding_slot(&target_slot, alloc);
+
+        log::debug!(
+            "\tTRANS: from='{:?}' to='{:?}' value='{:?}'",
+            self.source.firm,
+            self.target.firm,
+            target_slot.firm,
+        );
+        match &*source_slot {
+            MultiSlot::Single(slot) => assert_eq!(slot.firm, target_slot.firm),
+            MultiSlot::Multi { phi, .. } if Node::Phi(*phi) == target_slot.firm => (),
+            MultiSlot::Multi { slots, .. } => assert!(
+                slots.iter().any(|slot| slot.firm == target_slot.firm),
+                "{:?} does not contain slot with firm == {:?}",
+                slots.iter().map(|slot| slot.firm).collect(): Vec<_>,
+                target_slot.firm
+            ),
+        }
+
+        self.register_transitions.push((source_slot, target_slot));
+    }
+
+    /// Do there exist multiple incoming flows for the target slot of `flow_idx`
+    /// in the target block?
+    pub fn must_copy_in_source(self, flow_idx: usize) -> bool {
+        let target_slot_num = self.register_transitions[flow_idx].1.num;
+
+        self.target
+            .preds
+            .iter()
+            .filter(|pred| !Ptr::ptr_eq(**pred, self))
+            .any(|pred| {
+                self.register_transitions
```

Comment:
```suggestion
                pred.register_transitions
```

Shouldn't this still be pred?

https://github.com/comprakt/comprakt/pull/172/commits/4c1d43afd7689e17cb1f751b6980bb6db9d30e5c#diff-cadce81ee10b6d85cbc3a352126faebfR754

Reply:
probably, was already wondering about the unused variable warning